text/0060-bazel.md

```diff
@@ -0,0 +1,137 @@
+        ":aws-amazonmq",
+    ],
+)
+```
+
+The `pkg_npm` rule is also built-in to Bazel, and bundles a package based on given sources, along with other needed files like a package.json.
+Note that the `cdk_library` rule is only needed for packages that have a CFN to TS transformation. If that's not needed, it's simply a `ts_library` rule.
+
+# Drawbacks
+
+There are two main drawbacks to migrating to Bazel:
+
+1. Lack of familiarity with Bazel amongst the team
+1. Relative newness of Bazel in the JavaScript/TypeScript ecosystem
+
+For the first point, the current system has the advantage of being a glorified monorepo wrapper around NPM. Bazel,
+on the other hand, has a completely different semantic structure. While using it is straightforward for the most part,
+understanding how it works and more importantly how to extend it (by writing custom rules, for example) is not.
+
+For the second point, while Bazel has had its 1.0 release, that release was relatively recent, and there is still much
+work to be done in the space. For instance, Bazel does not currently supported nested package.json dependencies when
+constructing its own dependency tree. While the nested files can exist, their dependencies would have to be deduped to
+the root package.json file in order for everything to be setup properly by Bazel.
+
+# Rationale and Alternatives
+
+## Rationale
+* Bazel is incremental, meaning that only the changed aspects of the project and its dependencies need to be rebuilt
+* Bazel caches all of its builds, making it much faster than the current solution
+* Bazel allows for hermiticity and reproducability since it runs in a sandbox
+* Bazel is highly extensible, meaning custom rules can be written solely in TypeScript/JavaScript as opposed to Bash, but
+still incorporated easily into the toolchain
+
+## Alternatives
+* Remain with the current build system, but incorporate [lerno](https://github.com/aws/aws-cdk/pull/5359)
+
+# Adoption Strategy
+
+The Bazel rules can be added in parallel with current efforts by the team. None of the resources involved would
+overwrite or interfere with the current project structure, so adding files would not interfere with developer flow.
+Switching over to Bazel could also be opt-in as people want to try it, with the legacy system in place to manage
+releases and for developers to fallback on if needed.
+
+The rules for each subpackage would need to be added, along with the general rules for the project. All of the Bazel
+dependencies would be added to the top-level package.json, in addition to the nested dependencies from all of the
+subpackages. There should not be too many, since most of the "dependencies" are simply other CDK subpackages.
+
+# Unresolved questions
+
+* Will the project structure of the CDK remain stable over time, or will it move to a singular import?
+
+# Future Possibilities
+
+* We could setup Remote Build Execution (RBE) using a fleet on EC2 or ECS, further speeding up the team's builds
```

Comment:
What does that mean?

Reply:
It means that instead of using your local machine to execute build and test tasks, your Bazel instance would "outsource" the job to a team-available fleet (only accessible by trusted contributors, of course). Since Bazel is parallelizable, the work for a large and complex build task like the CDK could be spread out over 3 or 4 EC2 hosts and return the result to you locally in a fraction of the time it would otherwise take.